File: csharp/demo/demo09-detach.cs
/*
 * THREAD DETACHING
 */
using System;
using System.Threading;



class Demo09 : IRunnable
{
    public void run()
    {
        var thFoo = new Thread(() => {
            Console.WriteLine("foo is starting...");
            Thread.Sleep(2000);
            Console.WriteLine("foo is exiting...");
        });


        thFoo.IsBackground = true;
        thFoo.Start();


        // If I comment this statement,
        // thFoo will be forced into terminating with main thread
        Thread.Sleep(3000);


        Console.WriteLine("Main thread is exiting");
    }
}

File: python/demo09_detach.py
'''
THREAD DETACHING
'''

import time
import threading



def do_task():
    print('foo is starting...')
    time.sleep(2)
    print('foo is exiting...')



th_foo = threading.Thread(target=do_task, daemon=True)
th_foo.start()

# If I comment this statement,
# th_foo will be forced into terminating with main thread
time.sleep(3)

print('Main thread is exiting')
